feat: more detailed ClassLib info for VB (recreated) (#8503)
unoplatform/uno: doc/articles/faq.md

Much like the new UI technologies from Microsoft, Uno Platform doesn’t support creation of new applications using VB.NET.

> If you have an existing VB.NET application which you would like to port/modernize for cross-platform scenarios with Uno Platform, you should be able to reuse all of your VB.NET business logic, packaged as NET standard libraries into a new Uno Platform app where only the new UI code would be defined in XAML with some glue in C#.

Reviewer: ```suggestion
If you have an existing VB.NET application which you would like to port/modernize for cross-platform scenarios with Uno Platform, you should be able to reuse all of your VB.NET business logic. I needs to be built as .NET Standard libraries, then used in a new Uno Platform app where only the new UI code would be defined in XAML with some glue in C#.
```

Author: Ok, although this comment is mis-addressed :)  Your remark is, in fact, to duke7553, as this sentence is from his commit dated Jan 12.

Reviewer: No worries. Let's apply it anyways :)

> To be exact, add "Class Library" VB project (not "Class Library (.Net Framework)", and not "Class Library (Universal Windows)"). When you want to use this library for UWP app compatible with phones (i.e., app that works on Windows Phones as Lumia), you have to change Target Framework to ".Net Standard 1.4" (in Project:Properties), as this is last .Net Standard that can be used with projects targetting Windows 10 15063. If you are targetting Windows 10 16299+, use .Net Standard 2.0. Then, in Uno heads for all platforms, add reference to this library (simplest way is to right click on "References" nodes inside these heads). You can use same Class Library also in your original, VB project. The same Class Library can also be used MAUI project, if you ever decide to port app from Uno to MAUI.

Reviewer 2: I don't think it is necessary to mention MAUI as it just makes it more confusing (usually it is a FAQ itself - does Uno run on top of MAUI?). It gets even more confusing as we offered WebAssembly renderers for XF.

Author: As "an answer" to your "confusing ...  does Uno run on top of MAUI", I added "port app from Uno to MAUI". So, it is clear that Uno doesn't run on top of MAUI.
No-one knows what will last longer - MAUI, or Uno. Maybe MAUI will be abandoned by Microsoft, similarly to abandoning VB (I remember days when Microsoft claims "if you want to write programs for Windows, use VB, it is the best way for Windows programming"), or abandoning UWP ("best platform, unification of app code for all platforms"). But maybe Uno would stale, without new PRs. But any way, Class Library would work. it should work in any other future scenarios also.
So, mentioning MAUI is an argument: use Class Library, porting code to it would not be dead-end, your time invested in porting code to ClassLib would be sound investment in any future can be thought of.

> To be exact, add "Class Library" VB project (not "Class Library (.Net Framework)", and not "Class Library (Universal Windows)"). When you want to use this library for UWP app compatible with phones (i.e., app that works on Windows Phones as Lumia), you have to change Target Framework to ".Net Standard 1.4" (in Project:Properties), as this is last .Net Standard that can be used with projects targetting Windows 10 15063. If you are targetting Windows 10 16299+, use .Net Standard 2.0. Then, in Uno heads for all platforms, add reference to this library (simplest way is to right click on "References" nodes inside these heads). You can use same Class Library also in your original, VB project. The same Class Library can also be used MAUI project, if you ever decide to port app from Uno to MAUI.

Reviewer: ```suggestion
Furthermore, you'll need to add a `Class Library` Visual Basic project (not a `Class Library (.NET Framework)` or `Class Library (Universal Windows)`). 

While now unsupported by Microsoft, if you want to use this library for UWP app compatible with Windows Phone (i.e., app that works on a Lumia), you have to change Target Framework to ".NET Standard 1.4" (in Project, Properties), as this is last .NET Standard version which can be used with projects targeting Windows 10 15063. If you are targeting Windows 10 16299 or later, use .NET Standard 2.0. To use this library in Uno heads for all platforms, add a reference to this library (the simplest way is to right click on "References" nodes inside these heads).

You can also use same Class Library in your original, Visual Basic project. The same Class Library can also be used in any other .NET Standard compatible projects.
```

Reviewer: Sure. The rest still applies.

> Additionally, If you’d like to move any of your VB.NET code to C# you may be able to use automated tools such as https://converter.telerik.com

Reviewer: ```suggestion
Additionally, if you’d like to move any of your VB.NET code to C# you may be able to use automated tools such as https://converter.telerik.com.
```
